### ISSUE TITLE
implement [COPY] button for summary table?

### ISSUE THREAD

**Issue by edeutsch** [RTXteam/RTX#532]
In the summary table of the RTX output, copy pasting into Excel doesn't work that well. The column headers have the [+A] [+B] business, etc. Maybe it would be easy to have a single large friendly button at the top [COPY to Clipboard] or something that triggers a little js function that copies the contents of the table to the windows clipboard mostly sanitized, perhaps just a TSV string that can paste into Excel very tidily?

**Comment by isbluis** [comment]
Is this still desired functionality?  I hardly see the summary table returned with queries as of late, but perhaps there is a plan to populate it more often?

**Comment by edeutsch** [comment]
Still desirable eventually, but as you say, current code is not populating the table anymore. But we should. But it is a much harder problem. Back when we had canned queries, it was relatively straightforward to "pre-can" how to extract the most relevant information for the table. Now that there's no pre-canned but rather wild-west flexibility, it's harder to know what subset to put in the table. Doable, just not done.

So I would put this issue way low on the priority list.

**Comment by isbluis** [comment]
Perhaps the table could simply have the values of `essence`, if any, as a start?  And also `score`, once it is more widely reported?
Unless not terribly useful or relevant.

**Comment by edeutsch** [comment]
yes, a good idea. and some new code I added makes this pretty easy.
I have just rolled out the code everywhere. It is probably brittle and buggy, but give it a try and report!
I have vague plans to make the table more interesting soon, but for now, this probably includes everything you need for further development of the UI.

**Comment by isbluis** [comment]
Hi @edeutsch .

1. I can move the button to the top, though I think it breaks the page layout somewhat.

2. I cannot reproduce this in over a dozen attempts.  I get exactly one copy of the table no matter what the previous state of my buffer, and/or how many times I click on [copy].  Can you provide the steps that result in this behavior?  Or perhaps you have some special buffer utility installed on your system that does this? Or maybe browser-dependent...?

**Comment by edeutsch** [comment]
Repro on my environment:

- Firefox
- https://arax.rtx.ai/test/
- [SHIFT+RELOAD]
- [ENG] Tab
- Q3: What proteins does acetaminophen target?
- [ASK]
- [Summary]
- [Copy Summary table to clipboard (TSV)]
- [Query]
- [DSL] Tab
- Paste:
add_qnode(name=lovastatin, id=n0)
add_qnode(type=protein, id=n1)
add_qedge(source_id=n0, target_id=n1, id=e0)
expand(edge_id=e0)
overlay(action=compute_ngd, virtual_relation_label=N1, source_qnode_id=n0, target_qnode_id=n1)
resultify()
- [POST]
- [Summary]
- [Copy Summary table to clipboard (TSV)]
- Open Excel
- Paste
I get 63 lines, both tables appended

**Comment by isbluis** [comment]
Put in a fix (and moved button).
Please verify in devLM/ when you have a moment and I'll check it in.

**Comment by edeutsch** [comment]
Looks great to me! Push it!

**Comment by isbluis** [comment]
Pushed e95094d
